pkg/vault/test/foundry/fork/YieldBearingPoolSwapBase.t.sol

```diff
@@ -917,118 +912,105 @@ abstract contract YieldBearingPoolSwapBase is BaseVaultTest {
     }
 
     function _previewWrapExactOut(
         IERC4626 wToken,
         uint256 amountOutWrapped,
         bool withBufferLiquidity
-    ) private returns (uint256 amountInUnderlying, uint256 bufferUnderlyingSurplus, uint256 bufferWrappedSurplus) {
-        if (withBufferLiquidity) {
-            // If operation is within buffer range, convert is used by the vault to save some gas. Surpluses are 0,
-            // since there's no interaction with the wrapper protocol (therefore, no rebalance).
-            amountInUnderlying = wToken.previewMint(amountOutWrapped);
-        } else {
+    )
+        private
+        view
+        returns (uint256 amountInUnderlying, int256 bufferUnderlyingImbalance, int256 bufferWrappedImbalance)
+    {
+        amountInUnderlying = wToken.previewMint(amountOutWrapped);
+        bufferUnderlyingImbalance = 0;
+        bufferWrappedImbalance = 0;
+
+        // If operation is out of buffer liquidity, we need to wrap the underlying tokens in the wrapper protocol. But,
+        // if the buffer has enough liquidity, return the amountInUnderlying calculated by previewMint and imbalance 0.
+        if (withBufferLiquidity == false) {
             bytes32 bufferBalances = vault.getBufferTokenBalancesBytes(wToken);
-            // Mint converts underlying to wrapped. If buffer has a surplus of underlying, the vault wraps it to
-            // rebalance the buffer. It can introduce rounding issues, so we should consider the rebalance in our
-            // result preview. The logic below reproduces Vault's rebalance logic.
-            bufferUnderlyingSurplus = bufferBalances.getBufferUnderlyingSurplus(wToken);
-            bufferWrappedSurplus = wToken.previewDeposit(bufferUnderlyingSurplus);
-
-            // Do the actual operation to impact the rate used to calculate underlying surplus.
-            uint256 snapshotId = vm.snapshot();
-            vm.startPrank(lp);
-            uint256 vaultWrappedDeltaHint = amountOutWrapped + bufferWrappedSurplus;
-            IERC20(wToken.asset()).forceApprove(address(wToken), wToken.previewMint(vaultWrappedDeltaHint));
-            uint256 vaultUnderlyingDeltaHint = wToken.mint(vaultWrappedDeltaHint, lp);
-            vm.stopPrank();
-            bufferUnderlyingSurplus = wToken.previewMint(bufferWrappedSurplus);
-            vm.revertTo(snapshotId);
-
-            amountInUnderlying = vaultUnderlyingDeltaHint - bufferUnderlyingSurplus;
+            // Mint converts underlying to wrapped. The rebalance logic can introduce rounding issues, so we
+            // should consider it in our result preview. The logic below reproduces Vault's rebalance logic.
+            int256 bufferImbalance = bufferBalances.getBufferWrappedImbalance(wToken);
+            uint256 vaultWrappedDeltaHint = uint256(int256(amountOutWrapped) - bufferImbalance);
+            uint256 vaultUnderlyingDeltaHint = wToken.previewMint(vaultWrappedDeltaHint);
+
+            if (bufferImbalance != 0) {
```

Comment:
do these `if`s still matter now?

Reply:
I left them to make sure the behavior of the buffer is the same as before, even removing the last `if` from the vault code.